external/easylogging++/easylogging++.cc

```diff
@@ -2191,16 +2191,11 @@ void VRegistry::setFromArgs(const base::utils::CommandLineArgs* commandLineArgs)
 #   define ELPP_DEFAULT_LOGGING_FLAGS 0x0
 #endif // !defined(ELPP_DEFAULT_LOGGING_FLAGS)
 // Storage
-el::base::type::StoragePointer getresetELPP(bool reset)
+el::base::type::StoragePointer &el::base::Storage::getELPP()
 {
-  static el::base::type::StoragePointer p(new el::base::Storage(el::LogBuilderPtr(new el::base::DefaultLogBuilder())));
-  if (reset)
-    p = NULL;
-  return p;
-}
-el::base::type::StoragePointer el::base::Storage::getELPP()
-{
-  return getresetELPP(false);
+  if (!el::base::elStorage)
+    el::base::elStorage = new el::base::Storage(el::LogBuilderPtr(new el::base::DefaultLogBuilder()));
```

Comment:
At least one log is called rom a static object ctor (crypto::null_skey -> mlock -> get_page_size), so efore starting threads.
"allowed" is still showing a bit on the profile, I suspect just the locking, but that's still acceptable now. getresetELPP was just > 2%, whch seemed outrageous.

Reply:
Relying on a static object to be constructed is problematic - the process could be linked without the crypto lib. Additionally, I hope to provide a patch that removes that log before main because it is impossible to disable logging to console (useful for the light wallet server admin process which can be used by python apps or bash scripts etc).

Reply:
BTW - this _probably_ won't cause a problem, but I'm not sure how helpful the 2% in the profile is in this case - there are log statements everywhere so this function is going to get skewed due to all of those calls. How much on aggregate do some of these functions with log statements perform now? It should be a fairly small increase overall.

Reply:
I don't understand what you're asking.